databricks/koalas/series.py

```diff
@@ -4383,6 +4383,12 @@ def __len__(self):
         return len(self.to_dataframe())
 
     def __getitem__(self, key):
+        if isinstance(key, Series):
+            bcol = key._scol.cast(BooleanType())
+            sdf = self._internal.sdf.filter(bcol)
```

Comment:
Also, ideally we should also consider about different Series from different DataFrames.

We can do it via using existing logic that handles such series from different DataFrames:

```python
// To allows 'compute.ops_on_diff_frames', it assigns the given column to the current Frame.
self._kdf["__temp_col__"] = key
sdf = self._internal.sdf.filter(F.col("__temp_col__"))
internal = self._internal.copy(sdf=sdf)
...
```

^^ It's not tested so should be double checked.

Reply:
Thanks for the comment. Does the code above work for both different and same dataframes?

Reply:
Yeah, I think so because I used such similar trick (e.g., https://github.com/databricks/koalas/commit/7070bc6e29dd2ae628b51efa2972112e6b833bff#diff-5ba644c40e914732959b9b4867f30c53R2158)

Reply:
We could add one test under `OpsOnDiffFramesEnabledTest` (e.g., `kser[ks.Series([True])]`)

Reply:
Oh! wait, can we reuse `where`?

Reply:
Let me try

Reply:
`self.to_dataframe().where(...)` didn't work. It just fills `False` with `NaN`.